### PR TITLE
Mule 14423: Refactor aggregators-extension to propagate variables

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/runtime/privileged/ExecutionContextProperties.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.api.runtime.privileged;
+
+
+import static org.mule.runtime.core.api.config.MuleProperties.PROPERTY_PREFIX;
+import org.mule.runtime.extension.api.runtime.process.CompletionCallback;
+
+public class ExecutionContextProperties {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/runtime/privileged/ExecutionContextProperties.java]: jdoc

File: tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java [hunk 2]
```diff
@@ -27,21 +27,22 @@
...
+  private static final int POLLING_DELAY = 500;
 
   private static final String EXECUTION_ROUTE_KEY = "executionRoute";
 
-  private static Map<String, ExecutionLog> executionLogsMap = new HashMap<>();
+  private static Map<String, ExecutionLog> executionLogsMap = new ConcurrentHashMap<>();
 
   public static void resetLogsMap() {
-    executionLogsMap = new HashMap<>();
+    executionLogsMap = new ConcurrentHashMap<>();
```
Comment (alegmarra) [inline tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java]: why was this change needed?

File: tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java [hunk 3]
```diff
@@ -27,21 +27,22 @@
...
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Saves the payload that goes through a processor for accessing it later
  */
 public class FlowExecutionLogger extends AbstractComponent implements Processor {
 
   private static final int POLLING_TIMEOUT = 5000;
-  private static final int POLLING_DELAY = 0;
+  private static final int POLLING_DELAY = 500;
```
Comment (alegmarra) [inline tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java]: does this affect tests other than the aggregators?

File: tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java (hunk 2 above)
Comment (lucianoRM) [reply tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java]: The aggregation-complete and aggregator-listener were executed concurrently and both were writing on the same at the same time.
